addons/website_sale/models/product_product.py

```diff
@@ -100,6 +100,7 @@ def _get_contextual_price_tax_selection(self):
         self.ensure_one()
         price = self._get_contextual_price()
         line_tax_type = self.env['ir.config_parameter'].sudo().get_param('account.show_line_subtotals_tax_selection')
-        if line_tax_type == "tax_included" and self.taxes_id:
-            price = self.taxes_id.compute_all(price, product=self, partner=self.env['res.partner'])['total_included']
+        company_taxes = self.taxes_id.filtered(lambda tax: tax.company_id.id in self._context.get('allowed_company_ids'))
```

Comment:
```suggestion
        company_taxes = self.taxes_id.filtered(lambda tax: tax.company_id == self.env.company)
```

the context key `allowed_company_ids` should not be managed in business code unless really necessary.  Use the environment company instead (which encapsulates the `allowed_company_ids` logic and provides a fallback when the context key is not specified).

Reply:
I made the change to the commit. Thank you for the info!